drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -755,11 +755,72 @@ static int siwx91x_get_version(const struct device *dev, struct wifi_version *pa
 	return 0;
 }
 
+static int siwx91x_get_connected_ap_beacon_interval(void)
+{
+	sl_wifi_operational_statistics_t sl_stat = { };
+	sl_status_t status;
+
+	status = sl_wifi_get_operational_statistics(SL_WIFI_CLIENT_INTERFACE, &sl_stat);
+	if (status != SL_STATUS_OK) {
+		return status;
+	}
+
+	return sys_get_le16(sl_stat.beacon_interval);
+}
+
+static int siwx91x_convert_li_bcn_unit_to_ms(uint16_t bcn_li)
+{
+	uint16_t beacon_interval;
+	uint16_t ms_li;
+
+	beacon_interval = siwx91x_get_connected_ap_beacon_interval();
+	if (beacon_interval < 0) {
+		LOG_ERR("Failed to get beacon interval: 0x%x", beacon_interval);
+		return -EAGAIN;
+	}
+
+	ms_li = bcn_li * beacon_interval;
```

Comment:
If you want to use `uint16_t`, you have to add a comment explaining why you guarantee there is no overflow here.